File: frontend/src/api/intelligence.ts
import { apiClient } from './client';

export const getAthenaStatus = async () => {
  const { data } = await apiClient.get('/athena/status');
  return data;
};

export const getPlannerStatus = async () => {
  const { data } = await apiClient.get('/planner/status');
  return data;
};

export const getExecutorStatus = async () => {
  const { data } = await apiClient.get('/executor/status');
  return data;
};

export const getMemoryStatus = async () => {
  const { data } = await apiClient.get('/memory/status');
  return data;
};

File: backend/app/api/endpoints/intelligence.py
from fastapi import APIRouter

router = APIRouter(tags=["intelligence"])

@router.get("/athena/status")
def athena_status():
    return {"status": "healthy", "service": "athena"}

@router.get("/planner/status")
def planner_status():
    return {"status": "healthy", "service": "planner"}

@router.get("/executor/status")
def executor_status():
    return {"status": "healthy", "service": "executor"}

@router.get("/memory/status")
def memory_status():
    return {"status": "healthy", "service": "memory"}
